tools/update-dll-msvc.bat

```diff
@@ -18,90 +17,144 @@ rem \     'unix' : 'gmake',
 rem \    },
 rem \ }
 
-setlocal enabledelayedexpansion
-
-if "%PROCESSOR_ARCHITECTURE%"=="AMD64" (
-    set vimproc_arch=64
-) else (
-    set vimproc_arch=32
-)
+REM --------------------------------------------------------------------------------------------
+REM Set the default value for vimproc_arch and msvc_arch based on processor architecture.
+REM --------------------------------------------------------------------------------------------
+set vimproc_arch=32
+set msvc_arch=x86
+if /i "%PROCESSOR_ARCHITECTURE%"=="AMD64" set "vimproc_arch=64" & set "msvc_arch=x86_amd64" & goto Search_VC_Location
+if /i "%PROCESSOR_ARCHITEW6432%"=="AMD64" set "vimproc_arch=64" & set "msvc_arch=x86_amd64" & goto Search_VC_Location
```

Comment:
When running 32-bit Vim on 64-bit Windows, `%PROCESSOR_ARCHITECTURE%` is "x86" and `%PROCESSOR_ARCHITEW6432%` is "AMD64". In that case, 32-bit version of vimproc should be built.

Reply:
@k-takata , OK, I will fix it. Thanks for your suggestion.